src/Parsers/New/AST/INode.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <Parsers/New/AST/fwd_decl.h>
+
+#include <common/demangle.h>
+#include <Common/TypePromotion.h>
+#include <Parsers/ASTExpressionList.h>
+
+#include <initializer_list>
+#include <iostream>
+
+
+namespace DB::AST
+{
+
+class INode : public TypePromotion<INode>
+{
+    public:
+        virtual ~INode() = default;
+
+        virtual ASTPtr convertToOld() const { return ASTPtr(); }
+        virtual String toString() const { return {}; }
+
+        void dump() const { dump(0); }
+
+    protected:
+        INode() = default;
```

Comment:
compile error
> definition of implicit copy constructor for 'INode' is deprecated because it has a user-declared destructor [-Werror,-Wdeprecated]

so, we need add default copy constructor

`INode(const INode&) = default;`

Reply:
Please, provide your build configuration, we don't have this kind of errors in our CI builds.